airbyte-cdk/python/airbyte_cdk/sources/streams/concurrent/cursor.py

```diff
@@ -229,3 +231,17 @@ def ensure_at_least_one_state_emitted(self) -> None:
         called.
         """
         self._emit_state_message()
+
+    def generate_slices(self) -> Iterable[Tuple[Any, Any]]:
```

Comment:
We don't have the choice to implement this to fully rely on partitioned state. Still TODO:
* partition the generated slice using a slice_range
* ensure that the boundaries match the ones from the current Salesforce implementation